package/MDAnalysis/core/topologyobjects.py

```diff
@@ -116,7 +116,7 @@ def __hash__(self):
         return hash((self._u, tuple(self.indices)))
 
     def __repr__(self):
-        indices = (self.indices if self.indices[0] < self.indices[-1]
+        indices = np.sort(self.indices if self.indices[0] < self.indices[-1]
```

Comment:
Oops didn't add this to my review:

So looking at the initial issue, I think we want to here instead present the internal stored order of the data, not just sorting by indices. That way you know that your dihedral is in the right i,j,k,l order rather than whatever numeric sort returns.

Reply:
ah of course, thanks. Totally misunderstood the initial issue, makes sense now.